gears/src/types/address.rs

```diff
@@ -167,6 +184,44 @@ impl<const PREFIX: u8> From<BaseAddress<PREFIX>> for Vec<u8> {
     }
 }
 
+impl From<tendermint::types::proto::crypto::PublicKey> for ConsAddress {
```

Comment:
It is a question. 
Can we transform tendermint PublicKey into domain PublicKey? It allows us to generate any type of addresses.

Reply:
Tendermint has it's own domain PublicKey separate to Cosmos. I think it's cleaner to keep them separate. I'm not sure generating all the addresses makes sense for the two types of key e.g. I don't think a ConsAddress (consensus address) should be generated for a Cosmos PublicKey.